src/sentry/search/events/builder.py

```diff
@@ -1822,7 +1823,9 @@ def resolve_snql_function(
     def resolve_metric_index(self, value: str) -> Optional[int]:
         """Layer on top of the metric indexer so we'll only hit it at most once per value"""
         if value not in self._indexer_cache:
-            result = indexer.resolve(self.organization_id, value)
+            result = indexer.resolve(
+                self.organization_id, value, use_case_id=UseCaseKey.RELEASE_HEALTH
+            )
```

Comment:
This is discover code, should be performance usecase

Reply:
@ahmedetefy i assume we also need to fix it also in https://github.com/getsentry/sentry/blob/master/tests/sentry/search/events/test_builder.py#L778-L785, because string has "transaction" metrics in this specific test.